pkg/api/defaults.go

```diff
@@ -653,6 +655,8 @@ func (p *Properties) setAgentProfileDefaults(isUpgrade, isScale bool, cloudName
 				} else if profile.Distro == AKS1804Deprecated {
 					profile.Distro = AKSUbuntu1804
 				}
+			} else if p.OrchestratorProfile.KubernetesConfig.CustomHyperkubeImage != "" {
```

Comment:
we don't want to override a non-empty distro otherwise the instructions in the doc above to set it to 18.04 won't work... this needs to be inside the `if profile.Distro == "" && profile.ImageRef == nil` if statement

Reply:
yep